bootstrap/main.py

```diff
@@ -0,0 +1,130 @@
+
+default_file_path = pathlib.Path("/startup.json.default")
+config_file_path = pathlib.Path("/config/startup.json")
+
+CORE_CONTAINER_NAME = "companion_core"
+
+
+def ensure_dir(file_path: str) -> None:
+    """Makes sure that the file path exists
+
+    Args:
+        file_path (str): path to check/make
+    """
+    directory = os.path.dirname(file_path)
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def stop_old_container() -> None:
+    """ Looks for an old core container and stops it if found"""
+    try:
+        old_container = client.containers.get(CORE_CONTAINER_NAME)
+        old_container.stop()
+        old_container.remove()
+    except Exception:
+        pass
+
+
+def overwrite_config_file_with_defaults() -> None:
+    """Overwrites the config file with the default configuration"""
+    shutil.copy(default_file_path, config_file_path)
+
+
+def read_config_file() -> Any:
+    """Tries to read the config file
+
+    Returns:
+        Any: Json data of startup.json
+    """
+
+    # Tries to open the current file
+    try:
+        with open("/config/startup.json") as config_file:
+            return json.load(config_file)
+    except Exception:
+        # Copy defaults over and read again
+        shutil.copy(default_file_path, config_file_path)
+        with open("/config/startup.json") as config_file:
+            return json.load(config_file)
+
+
+def start_core() -> None:
+    """Loads core settings and launches the core docker. Loads default settings if no settings are found"""
+    core_version = "stable"
+
+    config = read_config_file()
+
+    try:
+        image = config["core"]["image"]
+        core_version = config["core"]["tag"]
+        binds = config["core"]["binds"]
+        ports = config["core"]["ports"]
+        privileged = config["core"]["privileged"]
+        network = config["core"]["network"]
+
+    except Exception as error:
+        print(f"Error reading startup json data! {error}")
+        sys.exit(1)
+
+    print("Starting core... this could take a while if the image is not available offline...")
+    client.containers.run(
+        f"{image}:{core_version}",
+        name=CORE_CONTAINER_NAME,
+        volumes=binds,
+        ports=ports,
+        privileged=privileged,
+        network=network,
+        detach=True,
+    )
+    print("Started, open http://localhost:8080")
+
+
+def core_is_running() -> bool:
+    """
+    Returns:
+        bool: True if the core container is running
+    """
+    for container in client.containers.list():
+        if CORE_CONTAINER_NAME in container.name:
+            return True
+    return False
+
+
+def remove_core() -> None:
+    """Deletes the core container if it exists (needed for updating the running image)"""
+    try:
+        old_container = client.containers.get(CORE_CONTAINER_NAME)
+        old_container.remove()
+    except Exception:
+        # This exception is raised if the container does not exist
```

Comment:
pass if it raises the exception if the container does not exist, and warn if a different exception is raised.